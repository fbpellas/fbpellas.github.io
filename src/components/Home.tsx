import * as React from 'react';
import { CardDeck } from 'react-bootstrap';
import { BASE_PATH_IMG } from '../constants';
import { Card } from './Card';

export interface HomeProps {
  setPageAndClear: (hash: string) => void;
}

const Home: React.FC<HomeProps> = (props) => {
  const { setPageAndClear } = props;
  return (
    <div className="block-2">
      <div className="article">
        <img
          className="full-img"
          src={`${BASE_PATH_IMG}IME Thesis.png`}
          alt="Learn the Art of Speaking American English"
        />
        <br />
        <CardDeck>
          <Card
            href='phonemes'
            title='Phonemes'
            description='Learn how to pronounce letters in English'
            setPageAndClear={setPageAndClear}
          />
          <Card
            href='stress'
            title='Word Stress'
            description='Understand how to emphasize each syllable'
            setPageAndClear={setPageAndClear}
          />
          <Card
            href='intonation'
            title='Intonation'
            description='Improve the pitch and the tone of your voice'
            setPageAndClear={setPageAndClear}
          />
        </CardDeck>
      </div>
    </div>
  );
}

export { Home }