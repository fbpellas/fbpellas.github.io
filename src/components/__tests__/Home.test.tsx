import React from 'react';
import { render } from '@testing-library/react';
import { Home } from '../Home';

test('Home', () => {
  const fn = jest.fn();
  const { container } = render(<Home setPageAndClear={fn} />);
  expect(container).toMatchSnapshot();
});
