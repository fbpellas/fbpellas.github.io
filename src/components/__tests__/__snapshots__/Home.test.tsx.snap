// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<div>
  <div
    class="block-2"
  >
    <div
      class="article"
    >
      <img
        alt="Learn the Art of Speaking American English"
        class="full-img"
        src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/IME Thesis.png"
      />
      <br />
      <div
        class="card-deck"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <p
              class="card-text"
            >
              Phonemes
            </p>
            <div
              class="card-title h5"
            >
              Learn how to pronounce letters in English
            </div>
            <a
              class="btn btn-secondary"
              href="#phonemes"
            >
              Learn
            </a>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <p
              class="card-text"
            >
              Word Stress
            </p>
            <div
              class="card-title h5"
            >
              Understand how to emphasize each syllable
            </div>
            <a
              class="btn btn-secondary"
              href="#stress"
            >
              Learn
            </a>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <p
              class="card-text"
            >
              Intonation
            </p>
            <div
              class="card-title h5"
            >
              Improve the pitch and the tone of your voice
            </div>
            <a
              class="btn btn-secondary"
              href="#intonation"
            >
              Learn
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
